src/router.test.ts

```diff
@@ -248,6 +255,76 @@ describe('type inference', () => {
         }),
     );
   });
+
+  test('type inference when using Zod transforms', async () => {
+    const clients = setup((router) =>
+      router
+        .declare({
+          name: 'getItems',
+          route: 'GET /items',
+          request: z.object({
+            // API should enforce a string, but the code should receive a number.
+            message: z.string().transform((val) => Number(val)),
+          }),
+          response: z.object({
+            // The code should return a number, and the API should return a string.
+            message: z.number(),
+          }),
+        })
+        .implement('GET /items', (ctx) => {
+          // this statement helps us validate that the message is typed as a Number
+          ctx.request.query.message.toFixed();
+          // This should compile -- it's enforcing a number.
+          return { message: 1 };
+        })
+        .declare({
+          name: 'createItem',
+          route: 'POST /items',
+          request: z.object({
+            // API should enforce a string, but the code should receive a number.
+            message: z.string().transform((val) => Number(val)),
+          }),
+          response: z.object({
+            // The code should return a number, and the API should return a string.
+            message: z.number(),
+          }),
+        })
+        .implement('POST /items', (ctx) => {
+          // this statement helps us validate that the message is typed as a Number
+          ctx.request.body.message.toFixed();
+          // This should compile -- it's enforcing a number.
+          return { message: 2 };
+        }),
+    );
+    // ---- GET VALIDATIONS ----
+
+    // Assert that client types enforce the API types.
+    const getInput: Parameters<typeof clients.typed.getItems>[0] = {} as any;
+    // @ts-expect-error This should fail -- the API requires a string
+    getInput.message = 1;
+
+    // This should compile -- the API requires a string.
+    const getResult = await clients.typed.getItems({ message: '1' });
+
+    // Confirm response is runtime correct.
+    expect(getResult.data).toStrictEqual({ message: 1 });
```

Comment:
Because of this comment above `// The code should return a number, and the API should return a string.` I thought we'd be validating that the API response here was a string. What am I missing?

Reply:
Nice catch. The comment is incorrect -- cruft from a previous version. I'll remove it in a follow-up unless there are other changes.